flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/table/BufferedUpsertSinkFunction.java

```diff
@@ -56,7 +56,10 @@
 /**
  * The wrapper of the {@link RichSinkFunction}. It will buffer the data and emit when the buffer is
  * full or timer is triggered or checkpointing.
+ *
+ * @deprecated Please use {@link ReducingUpsertSink} instead.
  */
+@Deprecated
 public class BufferedUpsertSinkFunction extends RichSinkFunction<RowData>
```

Comment:
Didn't you say that this guy is `@Public`? I don't see an explicit annotation and would say that it's defacto `@Internal` and can be removed.